mpi-proxy-split/mpi-wrappers/generate-mpi-fortran-wrappers.py

```diff
@@ -44,7 +44,7 @@ def print_mpi_type_struct():
   print("}")
 
 def print_mpi_type_hindexed():
-  print("#ifdef CRAY_MPICH_VERSION")
+  print("#if MPICH_NUMVERSION < MPICH_CALC_VERSION(3,4,0,0,2) && defined(CRAY_MPICH_VERSION)")
```

Comment:
Let's add a brief comment about what changed i MPICH-3.4.0.2

Reply:
Sure, I have added the comments.